contracts/implementation/Resolver.sol

```diff
@@ -14,18 +16,23 @@ contract Resolver is IResolver {
         poolKeeper = IPoolKeeper(poolFactory.getPoolKeeper());
     }
 
-    function upKeepChecker() external view override returns (bytes[] memory execPayLoad) {
+    /**
+     * @notice Iterate through all pools and will then return true if can be upkept, and data including the function selector and parameters.
+     * @return canExec True if pool requires upkeep
+     * @return execPayLoad payload data of the `isUpkeepRequiredSinglePool` function selector with the pool address as argument
+     */
+    function checker() external view override returns (bool canExec, bytes memory execPayLoad) {
         uint256 poolsLength = poolFactory.numPools();
-        execPayLoad = new bytes[](poolsLength);
         for (uint256 i = 0; i < poolsLength; ) {
```

Comment:
I know the increment occurs later (i.e., in the `unchecked` block) but I don't see the added value? If it's purely a gas optimisation endeavour then I don't think the loss of readability is justified. Happy for consenting opinion though @CalabashSquash .

Reply:
Yeah that's a fair point. I think this is only read off-chain actually so makes sense to run with the normie `for` loop.

Sorry for unnecessary change request.